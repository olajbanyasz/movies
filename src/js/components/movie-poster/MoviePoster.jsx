import React, { Component } from 'react';

class MoviePoster extends Component {
  render() {
    return (
      <img src={this.props.url}/>
    );
  }
}

export default MoviePoster;