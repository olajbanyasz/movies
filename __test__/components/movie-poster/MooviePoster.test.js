import React from 'react';
import { shallow } from 'enzyme';
import MoviePoster from '../../../src/js/components/movie-poster/MoviePoster';

describe('MoviePoster', () => {
  it('should rendered correctly', () => {
    const component = shallow(<MoviePoster url={'https://image.tmdb.org/t/p/w500/rPdtLWNsZmAtoZl9PK7S2wE3qiS.jpg'} />);
    expect(component).toMatchSnapshot();
  });
});
