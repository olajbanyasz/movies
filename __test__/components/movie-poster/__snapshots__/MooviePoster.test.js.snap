// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster should rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MoviePoster
    url="https://image.tmdb.org/t/p/w500/rPdtLWNsZmAtoZl9PK7S2wE3qiS.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "movie-poster",
      "src": "https://image.tmdb.org/t/p/w500/rPdtLWNsZmAtoZl9PK7S2wE3qiS.jpg",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "movie-poster",
        "src": "https://image.tmdb.org/t/p/w500/rPdtLWNsZmAtoZl9PK7S2wE3qiS.jpg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
